src/test/rustdoc-ui/intra-link-errors.rs

```diff
@@ -2,7 +2,16 @@
 //~^ NOTE lint level is defined
 
 // FIXME: this should say that it was skipped (maybe an allowed by default lint?)
-/// [<invalid syntax>]
+/// [invalid intra-doc syntax!!]
+
```

Comment:
Can you show what the error looks like when you try to use UFCS? I'm worried it will say something confusing.
```suggestion
/// FIXME(#74563) support UFCS
/// [<Vec as IntoIterator>::into_iter]
```

Reply:
Hmm, it's so-so. Any ideas on how to emit a better error?

```
error: unresolved link to `into_iter`
 --> src/test/rustdoc-ui/intra-link-errors.rs:8:6
  |
8 | /// [<Vec as IntoIterator>::into_iter]
  |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `into_iter` in `intra_link_errors`
  |
```

Reply:
I don't see a great way without implementing all of UFCS ... maybe you could see if one of the path segments a) has exactly two whitespace separators and b) has nothing outside the generics? At that point you're 90% of the way to implementing UFCS syntax though, just call `resolve()` with the path and `TypeNS`, then call [`filter_by_name`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.AssociatedItems.html#method.filter_by_name) on the trait to get the associated items it might match.

That might be better to leave to a later PR, though.

Reply:
Or what about just checking if it has ` as ` (with a space before and after) inside generics? I feel like that would be a decent heuristic.

Reply:
Sure, that seems reasonable.